src/jdk.jpackage/macosx/classes/jdk/jpackage/internal/MacPkgBundler.java

```diff
@@ -390,16 +390,20 @@ private String getRoot(Map<String, ? super Object> params,
         Path rootDir = appLocation.getParent() == null ?
                 Path.of(".") : appLocation.getParent();
 
-        Path[] list = Files.list(rootDir).toArray(Path[]::new);
-        if (list != null) { // Should not happend
-            // We should only have app image and/or .DS_Store
-            if (list.length == 1) {
-                return rootDir.toString();
-            } else if (list.length == 2) {
-                // Check case with app image and .DS_Store
-                if (list[0].toString().toLowerCase().endsWith(".ds_store") ||
-                    list[1].toString().toLowerCase().endsWith(".ds_store")) {
-                    return rootDir.toString(); // Only app image and .DS_Store
+        // Not needed for runtime installer and it might break runtime installer
+        // if parent does not have any other files
+        if (!StandardBundlerParam.isRuntimeInstaller(params)) {
+            Path[] list = Files.list(rootDir).toArray(Path[]::new);
```

Comment:
`Files.list()` requires try-with-resources construct.

Reply:
Fixed. I also removed check for null, since I do not think that Files.list can return null. I think it was leftover when we used File APIs instead of Files.